### ISSUE TITLE
python script for proxy server

### ISSUE THREAD

**Issue by quoehhh** [Soreepeong/XivMitmLatencyMitigator#60]
hello. and im back again with another problem xD

currently, im using my vps as forward proxy server. im using proxifier to send my game traffics to the vps(proxy server).
this is my best option with regards to gaming as i dont want to use VPN and it helps me alot on other mmorpg games

now upon checking the script, and how it works, (correct me if im wrong) it PREROUTES incoming traffic to the server's random socket, do the magic, and then routes it back to the internet.

with my vps setup, the server itself is the incoming traffic since he is acting as a proxy to forward requests to the game server.
and PREROUTING table will surely not work (also tried). I tried replacing PREROUTING to OUTPUT on the script, but the script just went on loop because im just REDIRECTING traffic endlessly on my vps.

tldr: how can i just make the script work without touching the iptables? do you have source python script that works on linux just like how it works on windows?

P.S. im a tech guy but with a little to no knowledge in socket programming. (still trying to learn it thou)

**Comment by quoehhh** [comment]
ok i found a band-aid solution.

I tried to same setup you did with hyper-v but using docker on my vps.
everything went smoothly until i end up here
![image](https://user-images.githubusercontent.com/74441901/191411726-caec3bb5-190e-4b9e-b237-ef8c026103ea.png)

i tried to ping the game-server IP but the results are this
![image](https://user-images.githubusercontent.com/74441901/191411814-4fea7308-6bc2-4514-9996-aa1734465404.png)

it feels like im still on a loop. client-vps-dockercontainer-vps-dockercontainer-vps-vps-dockercontainer-vps..........

this is the same setup you did with hyper-v. i dont know how you made it work? please help me here xD

**Comment by quoehhh** [comment]
FIXED:  made it work using docker container.
what i did was implement source based routing to segregate traffic going to the script and to actual game server to prevent routing loop.

**Comment by quoehhh** [comment]
Hello,

Currently my game traffic is routed via gameclient -> linuxVM -> container(contains the mitm script) -> linuxVM -> gameserver.

Here is my routing config on my linuxVM

#!/bin/sh
ip rule add from 172.21.20.206/32 table 14
ip route add 153.254.80.0/24 via 172.17.0.2 table 14
ip route add 124.150.157.0/24 via 172.17.0.2 table 14
ip route add 202.67.52.0/24 via 172.17.0.2 table 14
ip route add 204.2.229.0/24 via 172.17.0.2 table 14
ip route add 80.239.145.0/24 via 172.17.0.2 table 14
docker exec -i --privileged xivubuntu python3 mitigate.py

This was actually a script running as an autostartup service on my linuxVM.
1st line was my linuxVM interface IP. (im using WSL here btw)
2nd-6th line was game traffic coming out of container telling it to route it directly to game server IPs.
7th line was to start the container and run the script from there.

Hope this helps!
